tdrs-backend/tdpservice/search_indexes/admin/filters.py

```diff
@@ -0,0 +1,137 @@
+                selections.append(include)
+            if exclude and exclude in selections:
+                selections.remove(exclude)
+            if selections:
+                csv = ','.join(selections)
+                return changelist.get_query_string({self.parameter_name: csv})
+            else:
+                return changelist.get_query_string(remove=[self.parameter_name])
+
+        yield {
+            'selected': self.value() is None,
+            'query_string': changelist.get_query_string(remove=[self.parameter_name]),
+            'display': 'All',
+            'reset': True,
+        }
+        for lookup, title in self.lookup_choices:
+            yield {
+                'selected': str(lookup) in self.value_as_list(),
+                'query_string': changelist.get_query_string({self.parameter_name: lookup}),
+                'include_query_string': amend_query_string(include=str(lookup)),
+                'exclude_query_string': amend_query_string(exclude=str(lookup)),
+                'display': title,
+            }
+
+
+class CreationDateFilter(SimpleListFilter):
+    """Simple filter class to show newest created datafile records."""
+
+    title = _('Newest')
+
+    parameter_name = 'created_at'
+
+    def lookups(self, request, model_admin):
+        """Available options in dropdown."""
+        return (
+            (None, _('Newest')),
+            ('all', _('All')),
+        )
+
+    def queryset(self, request, queryset):
+        """Sort queryset to show latest records."""
+        if self.value() is None and queryset.exists():
+            datafiles = []
+            for record in queryset.order_by("datafile__stt__stt_code", "-datafile__id")\
+                    .distinct("datafile__stt__stt_code"):
+                datafiles.append(record.datafile)
+            return queryset.filter(datafile__in=datafiles)
+        return queryset
+
+
+class STTFilter(MultipleChoiceListFilter):
+    """Simple filter class to show records based on stt."""
+
+    title = _('STT Code')
+
+    parameter_name = 'stt_code'
+
+    def lookups(self, request, model_admin):
+        """Available options in dropdown."""
+        options = []
+        for obj in STT.objects.all():
+            options.append((obj.stt_code, _(obj.name)))
+        return options
+
+    def queryset(self, request, queryset):
+        """Return queryset of records based on stt code(s)."""
+        if self.value() is not None and queryset.exists():
+            stts = self.value().split(',')
+            queryset = queryset.filter(datafile__stt__stt_code__in=stts)
+        return queryset
+
+class FiscalPeriodFilter(SimpleListFilter):
+    """Simple filter class to filter records based on datafile fiscal year."""
+
+    title = _('Fiscal Period')
+
+    parameter_name = 'fiscal_period'
+
+    def lookups(self, request, model_admin):
+        """Available options in dropdown."""
+        current_year = datetime.date.today().year
```

Comment:
I would expect fiscal period on backend to be consistent with  what is available in the frontend dropdown. the minimum is FY2021, Q1.